signac/diff.py

```diff
@@ -0,0 +1,82 @@
+# Copyright (c) 2019 The Regents of the University of Michigan
+# All rights reserved.
+# This software is licensed under the BSD 3-Clause License.
+from .contrib.collection import _traverse_filter
+
+
+def _dotted_keys_to_nested_dicts(mapping):
+    """Converts dictionaries with dot-separated keys into nested dictionaries.
+
+    :param mapping: A mapping with dots in its keys, e.g. {'a.b': 'c'}
+    :returns: A mapping with nested keys, e.g. {'a': {'b': 'c'}}
+    :rtype: dict
+    """
+    result = {}
+
+    def make_nested_dict(d, keys):
+        item = d
+        for key in keys[:-1]:
+            if key not in item:
+                item[key] = {}
+            item = item[key]
+        return item
+
+    for dotted_key, value in mapping.items():
+        keys = dotted_key.split('.')
+        make_nested_dict(result, keys)[keys[-1]] = value
+
+    return result
+
+
+def diff_jobs(*jobs):
+    r"""Find differences among a list of jobs' state points.
+
+    The resulting diff is a dictionary where the keys are job ids and the
+    values are each job's state point minus the intersection of all provided
+    jobs' state points. The comparison is performed over the combined set
+    of keys and values.
+
+    Example:
```

Comment:
Please add an extra new line here. I'm not 100% sure if it renders fine without that, so better safe than sorry.

Reply:
This was indeed a bug -- I fixed it. You need newlines before _and_ after `.. code-block:: python`.